MuonAnalysis/MuonAssociators/src/PropagateToMuonSetup.cc

```diff
@@ -0,0 +1,81 @@
+#include "MuonAnalysis/MuonAssociators/interface/PropagateToMuonSetup.h"
+
+#include <cmath>
+
+#include "DataFormats/RecoCandidate/interface/RecoCandidate.h"
+
+#include "DataFormats/GeometrySurface/interface/TrapezoidalPlaneBounds.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "RecoMuon/Records/interface/MuonRecoGeometryRecord.h"
+#include "TrackingTools/DetLayers/interface/DetLayer.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+PropagateToMuonSetup::PropagateToMuonSetup(const edm::ParameterSet &iConfig, edm::ConsumesCollector iC)
+    : useSimpleGeometry_(iConfig.getParameter<bool>("useSimpleGeometry")),
+      useMB2_(iConfig.existsAs<bool>("useStation2") ? iConfig.getParameter<bool>("useStation2") : true),
```

Comment:
```suggestion
      useMB2_(iConfig.getParameter<bool>("useStation2")),
```
Just noting that it should be possible to remove the `existsAs` calls (which are generally discouraged, afaiu). One would have to set the proper defaults in `fillPSetDescription` (e.g. `add<bool>("useStation2", true)`, and so on).

(In any case, this part is under RECO, so not up to me to review, really).